### PR TITLE
Reenable tour command

### PR DESCRIPTION
**Pull request by didrocks**
Reenable "snapcraft tour" command with tests.

Added a small logic enhancements backed up with tests to ensure we don't nest default directory inside each other for multiple commands run.

### CODE REVIEW

**Comment by didrocks** [comment]
retest this please

**Comment by fgimenez** [comment]
retest this please

**Comment by fgimenez** [comment]
retest this please

File: snapcraft/tests/test_tour.py [hunk 1]
```diff
@@ -51,11 +49,13 @@ class TestScaffoldExample(TestCase):
     def setUp(self):
         super().setUp()
         self.temp_dir = None
+        self.cwd = os.getcwd()
 
     def tearDown(self):
         with suppress(OSError):
             if self.temp_dir:
                 shutil.rmtree(self.temp_dir)
+        os.chdir(self.cwd)
```
Comment (elopio) [inline snapcraft/tests/test_tour.py]: I missed this in the previous review.
We have a fixture for tmp cwd, which also works as a tempdir. Maybe you can replace your set up and tear down with that fixture, or inherit from the base test case.

**Comment by elopio** [comment]
all the commands need at least one integration test. That's not a blocker for this branch, but a blocker for releasing it. As always, I'm available to help.

File: snapcraft/tests/test_tour.py (hunk 1 above)
Comment (didrocks) [inline snapcraft/tests/test_tour.py]: Oh, I didn't see that base class add that fixture, indeed, using this and preventing creating other temp directory when that makes sense for the tests.

**Comment by didrocks** [comment]
pushed the base class changes + integration tests for the tour command.

As told on IRC, I didn't find all other commands beeing tested there, so I took the libery to create a new file for this command only and support 3 main tests cases. All other edge cases (copying to a directory which already has content, copying over a file…) are covered in unit tests.

File: integration_tests/test_tour.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import integration_tests
+import os
```
Comment (elopio) [inline integration_tests/test_tour.py]: first import os, then an empty line, then integration_tests

**Comment by elopio** [comment]
:100: Full unit coverage, and integration tests for the main cases, thank you!

**Comment by sergiusens** [comment]
retest this please
